logstash-core/src/main/java/org/logstash/secret/store/backend/JavaKeyStore.java

```diff
@@ -50,7 +50,7 @@
     private boolean useDefaultPass = false;
     private Lock writeLock;
     //package private for testing
-    static String filePermissions = "rw-rw----";
+    static String filePermissions = "rw-r--r--";
```

Comment:
(Comment, no change required) I might recommend simply 640 and having the file owned by `logstash` group, but this also works given the file is (intended to be) protected by a passphrase.

Reply:
I am a bit concerned at this level to make assumptions about the users on the local OS and the user that is running logstash.  With the package based installed we can make those assumptions, but for other types of installation that may be bad assumptions. 

This is a good point however, and look into incorporating this suggesting into the doc.